### PR TITLE
Fix mouse echo in UIA windows consoles when highlighter is on, avoid highlighter showing up when pressing alt+tab

### PR DESCRIPTION
**Pull request by LeonarddeR**
### Link to issue number:
None, reported in [this NVDA-devel thread](https://groups.io/g/nvda-devel/topic/61880059#44576)
Also a fix-up of #10520 

### Summary of the issue:
When the NVDA Highlighter is on, mouse echo is sometimes broken in UIA applications, particularly the windows console when UIA is on. This wasn't a bug in the highlighter, rather the following happens:

1. NVDA gets a window handle at the screen coordinates
2. NVDA checks for possible API classes. As the window handle of the console window is a native UIAElement, UIA is yielded as a possible API class.
3. UIA is used to get an UIAElement from the given point. However, it retrieves the UIAElement of the NVDA Highlighter window, which is an overlay. That window is non native.

Additionally, merging #10520 caused the highlighter to end up in the alt+tab order

### Description of how this pull request fixes the issue:
1. When finding out the object under the mouse, don't use UIA for non native UIA Elements.
2. Always recalculate the window handle when getting an UIA object from screen coordinates
3. Apply the WS_EX_TRANSPARENT style as requested in https://github.com/nvaccess/nvda/pull/10539#issuecomment-558035821
4. Apply WS_EX_NOACTIVATE to prevent the highlighter from showing up in the alt+tab order

### Testing performed:
* Tested that, when using the mouse in UIA consoles, the console text reads instead of the NVDA Highlighter window.
* Tested that the highlighter still shows up as normal. Tested visually by a sighted person.

### Known issues with pull request:
None known

### Change log entry:
While this first bug was there before the highlighter was introduced, I don't think an entry is needed as it was never discovered before.

### CODE REVIEW

**Comment by michaelDCurran** [comment]
Really IUIAutimation::elementFromPoint should not find the highlighter window at all. Does adding the WS_EX_TRANSPARENT extended window style to the highlighter window help at all here? My testing showed it stopped elementFromPoint picking it up, but I'm not sure if that causes any visual issues. I wouldn't think so as drawing in a transparent window still shows up. 
However, This bug also shows up another bug in that if the windows do differ between the window at point and UIA elementFromPoint, the original window is cached on the UIA NVDAObject, which means in the highlighter case, the windowClass was ConsoleWindowClass for the highlighter. That is pretty bad.
I would suggest that in UIA's kwargsFromSuper,  If the relation is a point, windowHandle should be removed from kwargs if we do insert UIAElement and return true. I agree we should return false if this is not a native element also.

**Comment by leonardder** [comment]
> Really IUIAutimation::elementFromPoint should not find the highlighter window at all. Does adding the WS_EX_TRANSPARENT extended window style to the highlighter window help at all here? My testing showed it stopped elementFromPoint picking it up, but I'm not sure if that causes any visual issues. I wouldn't think so as drawing in a transparent window still shows up.

I recall that WS_EX_TRANSPARENT causes issues for the highlighter, because we paint on the window itself, not on one of its beneath siblings. @feerrenrut, would you be able to test this by adding WS_EX_TRANSPARENT to HighlightWindow.extendedWindowStyle?

> However, This bug also shows up another bug in that if the windows do differ between the window at point and UIA elementFromPoint, the original window is cached on the UIA NVDAObject, which means in the highlighter case, the windowClass was ConsoleWindowClass for the highlighter. That is pretty bad.

> I would suggest that in UIA's kwargsFromSuper, If the relation is a point, windowHandle should be removed from kwargs if we do insert UIAElement and return true. I agree we should return false if this is not a native element also.

I've done that now.

I suggest merging as is and investigating the use of WS_EX_TRANSPARENT separately.

File: source/visionEnhancementProviders/NVDAHighlighter.py [hunk 1]
```diff
@@ -68,7 +68,12 @@ class HighlightWindow(CustomWindow):
 	className = u"NVDAHighlighter"
 	windowName = u"NVDA Highlighter Window"
 	windowStyle = winUser.WS_POPUP | winUser.WS_DISABLED
-	extendedWindowStyle = winUser.WS_EX_TOPMOST | winUser.WS_EX_LAYERED
+	extendedWindowStyle = (
+		winUser.WS_EX_TOPMOST
+		| winUser.WS_EX_LAYERED
+		| winUser.WS_EX_NOACTIVATE
```
Comment (feerrenrut) [inline source/visionEnhancementProviders/NVDAHighlighter.py]: Can you add comments here to say why we want these styles please.

**Review by feerrenrut — APPROVED** [review]
Thanks @leonardder

**Pull request merged**